---
title: "Figure1"
author: "Mick Adriaansens"
date: "`r Sys.Date()`"
output: html_document
editor_options: 
  chunk_output_type: console
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## R Markdown

This is an R Markdown document. Markdown is a simple formatting syntax for authoring HTML, PDF, and MS Word documents. For more details on using R Markdown see <http://rmarkdown.rstudio.com>.

When you click the **Knit** button a document will be generated that includes both content as well as the output of any embedded R code chunks within the document. You can embed an R code chunk like this:
#this file contains the number of genes per species, (not accounting for gene splicing), NCBI taxonomy and the taxonomic groups are called 'major tax'

#major tax is a more manual approach than phyla. Major tax issued over phyla, mainly due to ophistokonta as a phyla containing such diversity (Fungi, Metazoa, Choanoflagellata , etc). Similarly Archaeaplastida is also divided into Viridiplantae and Rhodophyta

#for each gene per species, if the gene count == 0 (0) or > 0 (1), a binary gene presence/absence is noted. 
```{r, make not in function}
'%!in%' <- function(x,y)!('%in%'(x,y))
library(dplyr)
```


```{r, Load Eukarya data and wrangel it a bit}
#these are the files containing all the antiporters, in our architecture it is called:
IT_Eukarya <- read.delim('/nesi/nobackup/uc04105/new_databases_May/Euk_database_May/IT_Eukarya_4nov.tsv', sep='\t', stringsAsFactors = TRUE)
#IT_Eukarya <- read.delim('MAdriaansens/Phylogeny_of_CPA/output/Antiporters_across_inhouseEukarya.tsv', sep='\t', stringsAsFactors = TRUE)



#this feels super round about, but as 'choanoflagellata is not a predefined character this is what needs to be done
IT_Eukarya$Major_tax <- as.character(IT_Eukarya$Major_tax)
#amorphaea also includes fungi and metazoa, so it not descriptive enough

IT_Eukarya$Major_tax[IT_Eukarya$Major_tax == 'Amorphea'] <- 'Choanoflagellata'
IT_Eukarya$Major_tax <- as.factor(IT_Eukarya$Major_tax)


IT_Eukarya$IT_count <- IT_Eukarya$NhaB_count + IT_Eukarya$NhaC_count + IT_Eukarya$NhaD_count
IT_Eukarya$Antiporter_count <- IT_Eukarya$CPA_count + IT_Eukarya$IT_count

IT_Eukarya$IT_binary[IT_Eukarya$IT_count == 0] <- 0
IT_Eukarya$IT_binary[IT_Eukarya$IT_count > 0] <- 1
#miss spelling
IT_Eukarya$Major_tax[IT_Eukarya$Major_tax == 'Hapophyta'] <- 'Haptophyta'


IT_Eukarya$Antiporter_binary[IT_Eukarya$Antiporter_count == 0] <- 0
IT_Eukarya$Antiporter_binary[IT_Eukarya$Antiporter_count >0] <- 1

#write.table(IT_Eukarya, file='Eukarya_7okt_edited.tsv', quote=FALSE, sep='\t')

```


```{r, get number per taxonomic group for gene numbers}
#now per major taxonomic group we count how many observations we got per group of species, genes and more
Major_tax_observations <- IT_Eukarya %>% count(Major_tax)
Major_tax_observations <- Major_tax_observations %>% rename('Species_total' = n)
#count NhaX/CPA genes per Major_tax
Major_tax_CPAobservations <- IT_Eukarya[IT_Eukarya$CPA_binary == 1,] %>%count(Major_tax)
Major_tax_CPAobservations<- merge(Major_tax_observations, Major_tax_CPAobservations, Major_tax = 'Major_tax')

#using this line of code I caclulate the percentage (lapply could do this faster but this is how I do it)
Major_tax_CPAobservations$percentage <-  (Major_tax_CPAobservations$n/Major_tax_CPAobservations$Species_total)*100


Major_tax_NhaBobservations <- IT_Eukarya[IT_Eukarya$NhaB_binary == 1,] %>%count(Major_tax)
Major_tax_NhaBobservations<- merge(Major_tax_observations, Major_tax_NhaBobservations, Major_tax = 'Major_tax')
Major_tax_NhaBobservations$percentage <-  (Major_tax_NhaBobservations$n/Major_tax_NhaBobservations$Species_total)*100

Major_tax_NhaCobservations <- IT_Eukarya[IT_Eukarya$NhaC_binary == 1,] %>%count(Major_tax)
Major_tax_NhaCobservations<- merge(Major_tax_observations, Major_tax_NhaCobservations, Major_tax = 'Major_tax')
Major_tax_NhaCobservations$percentage <-  (Major_tax_NhaCobservations$n/Major_tax_NhaCobservations$Species_total)*100
 
 
Major_tax_NhaDobservations <- IT_Eukarya[IT_Eukarya$NhaD_binary == 1,] %>%count(Major_tax)
Major_tax_NhaDobservations<- merge(Major_tax_observations, Major_tax_NhaDobservations, Major_tax = 'Major_tax')
Major_tax_NhaDobservations$percentage <-  (Major_tax_NhaDobservations$n/Major_tax_NhaDobservations$Species_total)*100
```

```{r, make_percentages_major_tax_euk}


Major_tax_A_no_AP_count <- aggregate(IT_Eukarya$Antiporter_binary==0, by=list(Major_tax=IT_Eukarya$Major_tax), FUN=sum)


IT_present <- IT_Eukarya[IT_Eukarya$IT_binary == 1,] 

#both IT and CPA
both <- aggregate(as.numeric(IT_present$CPA_binary), by=list(Major_tax=IT_present$Major_tax), FUN=sum)
both <- both %>% rename(both_CPA_IT=x)

#some major tax do all have IT
IT_absent <- IT_Eukarya[IT_Eukarya$IT_binary == 0,] 

no_IT_butCPA <- aggregate(as.numeric(IT_absent$CPA_binary), by=list(Major_tax=IT_absent$Major_tax), FUN=sum)
no_IT_butCPA <- no_IT_butCPA %>% rename(no_IT_butCPA=x)
#Species without CPA
CPA_absent <- IT_Eukarya[IT_Eukarya$CPA_binary == 0,] 

no_CPA_but_IT <- aggregate(CPA_absent$IT_binary, by=list(Major_tax=CPA_absent$Major_tax), FUN=sum)
no_CPA_but_IT  <- no_CPA_but_IT  %>% rename(no_CPA_but_IT =x)

Major_tax_A_no_AP_count <- aggregate(IT_Eukarya$Antiporter_binary==0, by=list(Major_tax=IT_Eukarya$Major_tax), FUN=sum)
Major_tax_A_no_AP_count <- Major_tax_A_no_AP_count %>% rename(total_noAP=x)

Major_tax_A_nothing_count <- aggregate(IT_Eukarya$Antiporter_binary==0, by=list(Major_tax=IT_Eukarya$Major_tax), FUN=sum)
Major_tax_A_nothing_count <- Major_tax_A_nothing_count %>% rename(no_AP=x)

Major_tax_A_AP_count <- aggregate(IT_Eukarya$Antiporter_count, by=list(Major_tax=IT_Eukarya$Major_tax), FUN=sum)
Major_tax_A_AP_count <- Major_tax_A_AP_count %>% rename(total_AP=x)

Major_tax_A_AP_species_count <- aggregate(IT_Eukarya$Antiporter_binary, by=list(Major_tax=IT_Eukarya$Major_tax), FUN=sum)
Major_tax_A_AP_species_count <- Major_tax_A_AP_species_count %>% rename(species_with_AP=x)
```


```{r, transform and combine dataframes}

Major_tax_CPA_species_count <- aggregate(IT_Eukarya$CPA_binary, by=list(Major_tax=IT_Eukarya$Major_tax), FUN=sum)
Major_tax_CPA_species_count <- Major_tax_CPA_species_count %>% rename(species_with_CPA=x)

Major_tax_A_CPA_count <- aggregate(as.numeric(IT_Eukarya$CPA_count), by=list(Major_tax=IT_Eukarya$Major_tax), FUN=sum)
Major_tax_A_CPA_count <- Major_tax_A_CPA_count %>% rename(total_CPA=x)

Major_tax_A_CPA_count <- merge(Major_tax_CPA_species_count, Major_tax_A_CPA_count, by='Major_tax')

Major_tax_A_AP_species_count <- aggregate(IT_Eukarya$Antiporter_binary, by=list(Major_tax=IT_Eukarya$Major_tax), FUN=sum)
Major_tax_A_AP_species_count <- Major_tax_A_AP_species_count %>% rename(species_with_AP=x)
```

```{r, generate figure 1C}
missing_tax = data.frame(Major_tax =c('Metazoa', 'Viridiplantae', 'Stramenopiles', 'Excavata','Amoebozoa', 'Alveolata', 'Apusomonadida', 'Rhodophyta', 'Cryptophyta', 'Choanoflagellata', 'Ancyromonadida', 'Filasterea', 'Ichthyosporea', 'Malawimonada', 'Haptophyta'),
                         no_CPA_but_IT=c(0, 0, 0,0,0,0,0,0,0,0,0,0,0, 0, 0))

no_CPA_but_IT <- rbind(no_CPA_but_IT, missing_tax)

missing_tax = data.frame(Major_tax =c('Apusomonadida', 'Rhodophyta', 'Choanoflagellata', 'Ancyromonadida', 'Excavata', 'Filasterea', 'Ichthyosporea', 'Malawimonada', 'Haptophyta', 'Rotosphaerida'),
                         no_IT_butCPA=c( 0, 0,0,0,0,0,0,0, 0,0))
no_IT_butCPA <- rbind(no_IT_butCPA, missing_tax)

#these small df will be used later down the line, even after the merge
Major_tax_A_accounts <- merge(Major_tax_A_no_AP_count, no_IT_butCPA, by='Major_tax')
Major_tax_A_accounts <- merge(Major_tax_A_accounts, both, by='Major_tax')
Major_tax_A_accounts <- merge(Major_tax_A_accounts, no_CPA_but_IT, by='Major_tax')
Major_tax_A_accounts <- merge(Major_tax_A_accounts, Major_tax_A_AP_count, by='Major_tax')
Major_tax_A_accounts <- merge(Major_tax_A_accounts, Major_tax_A_AP_species_count, by='Major_tax')
Major_tax_A_accounts <- merge(Major_tax_A_accounts, Major_tax_A_CPA_count, by='Major_tax')

Major_tax_A_accounts$IT_count <- Major_tax_A_accounts$total_AP - Major_tax_A_accounts$total_CPA

Major_tax_observations <- IT_Eukarya %>% count(Major_tax)
Major_tax_A_accounts <- merge(Major_tax_A_accounts, Major_tax_observations, by='Major_tax')
Major_tax_A_accounts$CPA_per_species <- Major_tax_A_accounts$total_CPA/Major_tax_A_accounts$species_with_AP
Major_tax_A_accounts$AP_per_species <- Major_tax_A_accounts$total_AP/Major_tax_A_accounts$species_with_AP

Major_tax_A_accounts$onlyCPA <- 100*Major_tax_A_accounts$no_IT_butCPA/Major_tax_A_accounts$n


Major_tax_A_accounts$bothCPA_and_IT <- 100*Major_tax_A_accounts$both_CPA_IT/Major_tax_A_accounts$n
Major_tax_A_accounts$onlyIT <- 100*Major_tax_A_accounts$no_CPA_but_IT/Major_tax_A_accounts$n

Percentages_Major_tax <- Major_tax_A_accounts[,c(1,14,15, 16)]
Percentages_Major_tax$Major_tax <- as.factor(Percentages_Major_tax$Major_tax)

pivot_percentages_MJT <- Percentages_Major_tax %>% pivot_longer(
  cols=-Major_tax,
  names_to = 'status',
  values_to = 'percentage') 

```

```{r generate_plot_Euk}
library(forcats)
#reorder the table to reflect the current Eukarya tree of life
order_burki_informed <- c('Stramenopiles', 'Alveolata', 'Rhizaria', 'Haptophyta', 'Cryptophyta', 'Viridiplantae', 'Rhodophyta', 'Apusomonadida', 'Fungi', 'Rotosphaerida', 'Ichthyosporea', 'Filasterea', 'Choanoflagellata','Metazoa','Amoebozoa', 'Excavata','Discoba', 'Metamonada', 'Malawimonada', 'Ancyromonadida' )

#burki order is informed based on Burki 2014 and Burki et al.2019
pivot_percentages_MJT$status <- factor(pivot_percentages_MJT$status, levels=c( 'onlyIT', 'bothCPA_and_IT', 'onlyCPA' ))

pivot_percentages_MJT$Major_tax <- factor(pivot_percentages_MJT$Major_tax, levels=order_burki_informed)

pivE<-ggplot(pivot_percentages_MJT, aes(y = Major_tax, x=percentage, fill=status, alpha=status))+ geom_bar(stat='identity') + scale_fill_manual(values = c('brown1', 'darkviolet', 'deepskyblue1')) + scale_alpha_manual(values = c(0.3, 0.9, 0.5)) + theme_minimal()  + 
  theme(axis.text.x = element_text(color = "black", size = 30, angle = 0, hjust = .5, vjust = .5, face = "plain"),
        axis.text.y = element_text(color = "black", size = 40, angle = 0, hjust = 1, vjust = 0, face = "bold"),  
        axis.title.x = element_text(color = "black", size = 20, angle = 0, hjust = .5, vjust = 0, face = "plain"),
        axis.title.y = element_text(color = "black", size = 25, angle = 0, hjust = .5, vjust = .5, face = "bold")) +theme(axis.text=element_text(size=30), legend.text = element_text(size=25)) + ggtitle('Eukarya') +theme(plot.title = element_text(size = 45, face = "bold")) +
  theme(plot.title = element_text(size = 50, face='bold' ),
        plot.subtitle = element_text(size = 20)) + theme(legend.text = element_text(size=40), legend.title = element_text(size=50, face='bold'))
ggsave("Eukarya_presence.pdf") 


```

```{r, create_subset_euks}
Major_tax_observations <- IT_Eukarya %>% count(Major_tax)
Major_tax_observations <- Major_tax_observations[Major_tax_observations$n != '1',]
```

```{r, make_group_only_species_with_both antiporters}
#we generate a set of percentages for 4 groups, Both IT and CPA, only CPA, only IT, and neither (which we do not call, but do generate by the sum of percentages)
Both_EONLY_SPECIES <- IT_Eukarya[IT_Eukarya$IT_count >0 & IT_Eukarya$CPA_count > 0 & IT_Eukarya$Major_tax %in% Major_tax_observations$Major_tax,]
print(Both_EONLY_SPECIES)

Observations_EBoth_only<- aggregate(as.numeric(Both_EONLY_SPECIES$Antiporter_count), by=list(Major_tax=Both_EONLY_SPECIES$Major_tax), FUN=sum)
Observations_EBoth_only <- Observations_EBoth_only %>% rename(
    Major_tax = Major_tax,
    total_Both=x)

Both_EONLY_Species_count <- Both_EONLY_SPECIES %>% count(Major_tax)
Both_EONLY_Species_count_per_phyla <- Both_EONLY_SPECIES %>% count(Major_tax)

rm(Both_EONLY_SPECIES)
rm(Both_EONLY_Species_count)

Both_EONLY_Species_count_per_phyla <- merge(Both_EONLY_Species_count_per_phyla, Observations_EBoth_only, phyla = 'Major_tax')
```

```{r, complete_cpa_only}
CPA_EONLY_SPECIES <- IT_Eukarya[IT_Eukarya$IT_count == '0' & IT_Eukarya$CPA_count > 0 & IT_Eukarya$Major_tax %in% Major_tax_observations$Major_tax,]


Observations_cpa_Eonly<- aggregate(as.numeric(CPA_EONLY_SPECIES$CPA_count), by=list(Major_tax=CPA_EONLY_SPECIES$Major_tax), FUN=sum)
Observations_cpa_Eonly <- Observations_cpa_Eonly %>% rename(
    Major_tax = Major_tax,
    total_CPA=x)

CPA_EONLY_Species_count <- CPA_EONLY_SPECIES %>% count(Major_tax)
CPA_EONLY_Species_count_per_phyla <- CPA_EONLY_SPECIES %>% count(Major_tax)

rm(CPA_EONLY_SPECIES)
rm(CPA_EONLY_Species_count)

CPA_EONLY_Species_count_per_phyla <- merge(CPA_EONLY_Species_count_per_phyla, Observations_cpa_Eonly, phyla = 'Major_tax')

'%!in%' <- function(x,y)!('%in%'(x,y))
Missing_phyla_list = list(c())
Missing_N_list = list(c())
Missing_gene_count = list(c())
for (taxa in Major_tax_observations$Major_tax) {

  if (taxa %!in% CPA_EONLY_Species_count_per_phyla$Major_tax) {
    print(taxa)
    Missing_phyla_list<-append(Missing_phyla_list, taxa)
    Missing_N_list<-append(Missing_N_list, 1000)
    Missing_gene_count<-append(Missing_gene_count, 0.001)
  }
}


Major_tax<- unlist(Missing_phyla_list, use.names=FALSE)
n<-unlist(Missing_N_list, use.names = FALSE)
total_CPA <-unlist(Missing_gene_count, use.names =FALSE)
Missingdata <- data.frame(Major_tax, n, total_CPA)

#E stands for Eukarya
CPA_EONLY_Species_count_per_phyla <- rbind(CPA_EONLY_Species_count_per_phyla, Missingdata)
```


```{r, Species with IT only}
IT_EONLY_SPECIES <- IT_Eukarya[IT_Eukarya$IT_count >0 & IT_Eukarya$CPA_count=='0' & IT_Eukarya$Major_tax %in% Major_tax_observations$Major_tax,]


Observations_IT_Eonly<- aggregate(as.numeric(IT_EONLY_SPECIES$IT_count), by=list(Major_tax=IT_EONLY_SPECIES$Major_tax), FUN=sum)
Observations_IT_Eonly <- Observations_IT_Eonly %>% rename(
    Major_tax = Major_tax,
    total_IT=x)

IT_EONLY_Species_count <- IT_EONLY_SPECIES %>% count(Major_tax)
IT_EONLY_Species_count_per_phyla <- IT_EONLY_SPECIES %>% count(Major_tax)

rm(IT_EONLY_SPECIES)
rm(IT_EONLY_Species_count)

IT_EONLY_Species_count_per_phyla <- merge(IT_EONLY_Species_count_per_phyla, Observations_IT_Eonly, phyla = 'Major_tax')

'%!in%' <- function(x,y)!('%in%'(x,y))
Missing_phyla_list = list(c())
Missing_N_list = list(c())
Missing_gene_count = list(c())
for (taxa in Major_tax_observations$Major_tax) {

  if (taxa %!in% IT_EONLY_Species_count_per_phyla$Major_tax) {
    print(taxa)
    Missing_phyla_list<-append(Missing_phyla_list, taxa)
    Missing_N_list<-append(Missing_N_list, 1000)
    Missing_gene_count<-append(Missing_gene_count, 0.001)
  }
}


Major_tax<- unlist(Missing_phyla_list, use.names=FALSE)
n<-unlist(Missing_N_list, use.names = FALSE)
total_IT <-unlist(Missing_gene_count, use.names =FALSE)
Missingdata <- data.frame(Major_tax, n, total_IT)

IT_EONLY_Species_count_per_phyla <- rbind(IT_EONLY_Species_count_per_phyla, Missingdata)
```

```{r, average numbers per taxa}
mergedE_all <-merge(Both_EONLY_Species_count_per_phyla,CPA_EONLY_Species_count_per_phyla, by='Major_tax')
mergedE_all <- merge(mergedE_all,IT_EONLY_Species_count_per_phyla, by='Major_tax')

mergedE_all <- mergedE_all %>% rename(
    n_species_onlyIT = n,
    n_species_both = n.x,
    n_species_onlyCPA = n.y)
mergedE_all$average_CPA <- mergedE_all$total_CPA/mergedE_all$n_species_onlyCPA
mergedE_all$average_Both <- mergedE_all$total_Both/mergedE_all$n_species_both
mergedE_all$average_IT <- mergedE_all$total_IT/mergedE_all$n_species_onlyIT
```


```{r}
Order_MJ <-  mergedE_all %>% arrange(desc(average_Both))
List_MJ <- Order_MJ$Major_tax
mergedE_all <- mergedE_all[c('Major_tax', 'average_CPA', 'average_Both', 'average_IT')]
pivot_Euk<- mergedE_all %>% pivot_longer(cols=-Major_tax, names_to = 'status',values_to = 'Count') 


MeanEuk <-  mean(IT_Eukarya$Antiporter_count[IT_Eukarya$Antiporter_binary==1 & IT_Eukarya$Major_tax != 'Stramenopiles' & IT_Eukarya$Major_tax != 'Cryptophyta'])

pivot_Euk$Major_tax <- factor(pivot_Euk$Major_tax, levels=c(List_MJ))
pivot_Euk$status <- factor(pivot_Euk$status, levels=c('average_CPA', 'average_Both', 'average_IT'))
PERE<-ggplot(pivot_Euk[pivot_Euk$Major_tax != 'Cryptophyta' & pivot_Euk$Major_tax != 'Stramenopiles',], aes(fill=status, y=Count, x=Major_tax)) + geom_bar(position = 'dodge', stat='identity') + theme_minimal()  + scale_fill_manual(values = c('deepskyblue1', 'darkviolet', 'brown1')) + scale_alpha_manual(values = c(0.3, 0.9, 0.5)) +
  theme(axis.text.x = element_text(color = "black", size = 45, angle = 90, hjust = .5, vjust = .5, face = "bold"),
        axis.text.y = element_text(color = "black", size = 20, angle = 0, hjust = 1, vjust = 0.5, face = "bold"),  
        axis.title.x = element_text(color = "black", size = 20, angle = 0, hjust = .5, vjust = 0, face = "plain"),
        axis.title.y = element_text(color = "black", size = 25, angle = 0, hjust = .5, vjust = .5, face = "bold")) + geom_vline(xintercept = c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5))  + geom_hline(yintercept = MeanEuk, lty='dashed') +
  scale_y_continuous(limits = c(0,40), expand = c(0, 0)) +
  labs(title = "Eukarya",
       subtitle = "Average antiporter per species grouped by antiporter presence"
       )+
  theme(plot.title = element_text(size = 50, face='bold' ),
        plot.subtitle = element_text(size = 20)) + theme(legend.text = element_text(size=40), legend.title = element_text(size=50, face='bold'))
PERE
```



```{r load_Archaea}
library(tidyr)
#these are the files containing all the antiporters, in our architecture it is called:
IT_Archaea <- read.delim('/nesi/nobackup/uc04105/new_databases_May/GTDB_226/IT_Archaea_7OKTSEPT.tsv', sep='\t', stringsAsFactors = TRUE)
#IT_Archaea <- read.delim('MAdriaansens/Phylogeny_of_CPA/output/Antiporters_across_ArchaeaGTDB226.tsv', sep='\t', stringsAsFactors = TRUE)




sum(IT_Archaea$CPA_count)
IT_Archaea %>% separate(GTDB_tax, c('Domain_Phyla', 'Lower_class'), ';c__') -> IT_Archaea
#split archaea taxonomy into different taxonomic groups
IT_Archaea %>% separate(Domain_Phyla, c('Domain', 'Phyla'), ';p__') ->IT_Archaea
IT_Archaea %>% separate(Lower_class, c('Class', 'Lower_order'), ';o__') -> IT_Archaea
IT_Archaea %>% separate(Lower_order, c('Order', 'Lower_family'), ';f__') -> IT_Archaea
IT_Archaea %>% separate(Lower_family, c('Family', 'Lower_genera'), ';g__') -> IT_Archaea
IT_Archaea %>% separate(Lower_genera, c('Genera', 'Species'), ';s__') -> IT_Archaea
IT_Archaea$IT_count <- IT_Archaea$NhaB_count + IT_Archaea$NhaC_count + IT_Archaea$NhaD_count
IT_Archaea$Antiporter_count <- IT_Archaea$CPA_count + IT_Archaea$IT_count

IT_Archaea$IT_binary[IT_Archaea$IT_count == 0] <- 0
IT_Archaea$IT_binary[IT_Archaea$IT_count > 0] <- 1

IT_Archaea$Antiporter_binary[IT_Archaea$Antiporter_count == 0] <- 0
IT_Archaea$Antiporter_binary[IT_Archaea$Antiporter_count > 0] <- 1

IT_Archaea$CPA_presence[IT_Archaea$CPA_binary ==0] <- 'Absent'
IT_Archaea$CPA_presence[IT_Archaea$CPA_binary ==1] <- 'Present'

IT_Archaea$IT_binary[IT_Archaea$IT_count == 0] <- 0
IT_Archaea$IT_binary[IT_Archaea$IT_count > 0] <- 1

TypesA <- read.delim('/nesi/nobackup/uc04105/new_databases_May/final_tree_set/Types_CPA_in_ArchaeaGTDB226.tsv', sep='\t', stringsAsFactors = TRUE)

IT_Archaea <- merge(IT_Archaea, TypesA, by ='GTDB_id', all =TRUE)
IT_Archaea[is.na(IT_Archaea)] <- 0
#generate which CPA types are present in which CPAs

rm(TypesA)
write.table(IT_Archaea, file='Antiporter_across_all_GTDB226_Archaea_30July_representatives.csv', quote=FALSE, sep=',')

```

```{r Taxonomic_/Protein_data_wrangling_Arc}
#generate a Archaea set of only high quality genomes

GTDB_Arc <- read.delim('/nesi/nobackup/uc04105/new_databases_May/GTDB_226/ar53_metadata.tsv', sep='\t', stringsAsFactors = TRUE, row.names = NULL)
GTDB_Arc$GTDB_id <- GTDB_Arc$accession


IT_Archaea <- merge(IT_Archaea, GTDB_Arc, by = 'GTDB_id')
rm(GTDB_Arc)

IT_Archaea <- IT_Archaea[IT_Archaea$Completeness> 90 & IT_Archaea$Contamination < 5 & IT_Archaea$trna_aa_count > 17,]

```








```{r, count number of species per phyla}

#count NhaX/CPA genes per phyla
PhylaA_observations <- IT_Archaea %>% count(Phyla)
PhylaA_observations <- PhylaA_observations %>% rename('Species_total' = n)
#CPA per Phyla
PhylaA_CPAobservations <- IT_Archaea[IT_Archaea$CPA_binary == 1,] %>%count(Phyla)
PhylaA_CPAobservations<- merge(PhylaA_observations, PhylaA_CPAobservations, phyla = 'Phyla')
PhylaA_CPAobservations$percentage <-  (PhylaA_CPAobservations$n/PhylaA_CPAobservations$Species_total)*100

#shows no NhaB in Archaea
PhylaA_NhaBobservations <- IT_Archaea[IT_Archaea$NhaB_binary == 1,] %>%count(Phyla)
PhylaA_NhaBobservations<- merge(PhylaA_observations, PhylaA_NhaBobservations, phyla = 'Phyla')
PhylaA_NhaBobservations$percentage <-  (PhylaA_NhaBobservations$n/PhylaA_NhaBobservations$Species_total)*100

#NhaC
PhylaA_NhaCobservations <- IT_Archaea[IT_Archaea$NhaC_binary == 1,] %>%count(Phyla)
PhylaA_NhaCobservations<- merge(PhylaA_observations, PhylaA_NhaCobservations, phyla = 'Phyla')
PhylaA_NhaCobservations$percentage <-  (PhylaA_NhaCobservations$n/PhylaA_NhaCobservations$Species_total)*100
 
#NhaD 
PhylaA_NhaDobservations <- IT_Archaea[IT_Archaea$NhaD_binary == 1,] %>%count(Phyla)
PhylaA_NhaDobservations<- merge(PhylaA_observations, PhylaA_NhaDobservations, phyla = 'Phyla')
PhylaA_NhaDobservations$percentage <-  (PhylaA_NhaDobservations$n/PhylaA_NhaDobservations$Species_total)*100
```





```{r, Only species with both Antiporters}
#we generate a set of percentages for 4 groups, Both IT and CPA, only CPA, only IT, and neither (which we do not call, but do generate by the sum of percentages)
Both_AONLY_SPECIES <- IT_Archaea[IT_Archaea$IT_count >0 & IT_Archaea$CPA_count > 0 & IT_Archaea$Phyla %in% PhylaA_observations$Phyla,]


Observations_ABoth_only<- aggregate(as.numeric(Both_AONLY_SPECIES$Antiporter_count), by=list(Major_tax=Both_AONLY_SPECIES$Phyla), FUN=sum)
Observations_ABoth_only <- Observations_ABoth_only %>% rename(
    Phyla = Major_tax,
    total_Both=x)

Both_AONLY_Species_count <- Both_AONLY_SPECIES %>% count(Phyla)
Both_AONLY_Species_count_per_phyla <- Both_AONLY_SPECIES %>% count(Phyla)



Both_AONLY_Species_count_per_phyla <- merge(Both_AONLY_Species_count_per_phyla, Observations_ABoth_only, phyla = 'Phyla')

#B1Sed does not have both, only has
```

```{r, complete_cpa_only}
CPA_AONLY_SPECIES <- IT_Archaea[IT_Archaea$IT_count == '0' & IT_Archaea$CPA_count > 0 & IT_Archaea$Phyla %in% PhylaA_observations$Phyla,]


Observations_cpa_Aonly<- aggregate(as.numeric(CPA_AONLY_SPECIES$CPA_count), by=list(Major_tax=CPA_AONLY_SPECIES$Phyla), FUN=sum)
Observations_cpa_Aonly <- Observations_cpa_Aonly %>% rename(
    Phyla = Major_tax,
    total_CPA=x)

CPA_AONLY_Species_count <- CPA_AONLY_SPECIES %>% count(Phyla)
CPA_AONLY_Species_count_per_phyla <- CPA_AONLY_SPECIES %>% count(Phyla)


CPA_AONLY_Species_count_per_phyla <- merge(CPA_AONLY_Species_count_per_phyla, Observations_cpa_Aonly, phyla = 'Phyla')

#new function
'%!in%' <- function(x,y)!('%in%'(x,y))


Missing_phyla_list = list(c())
Missing_N_list = list(c())
Missing_gene_count = list(c())
for (taxa in PhylaA_observations$Phyla) {

  if (taxa %!in% CPA_AONLY_Species_count_per_phyla$Phyla) {
    print(taxa)
    Missing_phyla_list<-append(Missing_phyla_list, taxa)
    Missing_N_list<-append(Missing_N_list, 1000)
    Missing_gene_count<-append(Missing_gene_count, 0.001)
  }
}


Phyla<- unlist(Missing_phyla_list, use.names=FALSE)
n<-unlist(Missing_N_list, use.names = FALSE)
total_CPA <-unlist(Missing_gene_count, use.names =FALSE)
Missingdata <- data.frame(Phyla, n, total_CPA)

CPA_AONLY_Species_count_per_phyla <- rbind(CPA_AONLY_Species_count_per_phyla, Missingdata)
```


```{r, IT homologs only Species}
IT_AONLY_SPECIES <- IT_Archaea[IT_Archaea$IT_count >0 & IT_Archaea$CPA_count=='0' & IT_Archaea$Phyla %in% PhylaA_observations$Phyla,]


Observations_IT_Aonly<- aggregate(as.numeric(IT_AONLY_SPECIES$IT_count), by=list(Major_tax=IT_AONLY_SPECIES$Phyla), FUN=sum)
Observations_IT_Aonly <- Observations_IT_Aonly %>% rename(
    Phyla = Major_tax,
    total_IT=x)

IT_AONLY_Species_count <- IT_AONLY_SPECIES %>% count(Phyla)
IT_AONLY_Species_count_per_phyla <- IT_AONLY_SPECIES %>% count(Phyla)

rm(IT_AONLY_SPECIES)
rm(IT_AONLY_Species_count)

IT_AONLY_Species_count_per_phyla <- merge(IT_AONLY_Species_count_per_phyla, Observations_IT_Aonly, phyla = 'Phyla')

'%!in%' <- function(x,y)!('%in%'(x,y))
Missing_phyla_list = list(c())
Missing_N_list = list(c())
Missing_gene_count = list(c())
for (taxa in PhylaA_observations$Phyla) {

  if (taxa %!in% IT_AONLY_Species_count_per_phyla$Phyla) {
    print(taxa)
    Missing_phyla_list<-append(Missing_phyla_list, taxa)
    Missing_N_list<-append(Missing_N_list, 1000)
    Missing_gene_count<-append(Missing_gene_count, 0.001)
  }
}


Phyla<- unlist(Missing_phyla_list, use.names=FALSE)
n<-unlist(Missing_N_list, use.names = FALSE)
total_IT <-unlist(Missing_gene_count, use.names =FALSE)
Missingdata <- data.frame(Phyla, n, total_IT)

IT_AONLY_Species_count_per_phyla <- rbind(IT_AONLY_Species_count_per_phyla, Missingdata)
```

```{r, merge and wrangle all datasets}
mergedA_all <-merge(Both_AONLY_Species_count_per_phyla,CPA_AONLY_Species_count_per_phyla, by='Phyla')
mergedA_all <- merge(mergedA_all,IT_AONLY_Species_count_per_phyla, by='Phyla')

mergedA_all <- mergedA_all %>% rename(
    n_species_onlyIT = n,
    n_species_both = n.x,
    n_species_onlyCPA = n.y)
mergedA_all$average_CPA <- mergedA_all$total_CPA/mergedA_all$n_species_onlyCPA
mergedA_all$average_Both <- mergedA_all$total_Both/mergedA_all$n_species_both
mergedA_all$average_IT <- mergedA_all$total_IT/mergedA_all$n_species_onlyIT
```



```{r, phyla with no antiporters}

Phyla_A_no_AP_count <- aggregate(IT_Archaea$Antiporter_binary==0, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)

Phyla_A_nothing_count <- aggregate(IT_Archaea$Antiporter_binary==0, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)

#percentage of species without antiporters, 9.52% miss an antiporter
100*sum(Phyla_A_nothing_count$x)/nrow(IT_Archaea)
Phyla_A_no_AP_count <- Phyla_A_no_AP_count %>% rename(total_noAP=x)

IT_present <- IT_Archaea[IT_Archaea$IT_binary == 1,] 

both <- aggregate(IT_present$CPA_binary, by=list(Phyla=IT_present$Phyla), FUN=sum)
both <- both %>% rename(both_CPA_IT=x)

IT_absent <- IT_Archaea[IT_Archaea$IT_binary == 0,] 

no_IT_butCPA <- aggregate(IT_absent$CPA_binary, by=list(Phyla=IT_absent$Phyla), FUN=sum)
no_IT_butCPA <- no_IT_butCPA %>% rename(no_IT_butCPA=x)


JACRDV01 <- c('JACRDV01', 0)
SpSt1190 <- c('SpSt-1190', 0)
Nanohalarchaeota <- c('Nanohalarchaeota', 0)
Altiarchaeota <- c('Altiarchaeota', 0)
Methanobacteriota_B <- c('Methanobacteriota_B', 0)
Undinarchaeoata <- c('Undinarchaeota', 0)
Korarchaeota <- c('Korarchaeota', 0)

B1Sed <- c('B1Sed10-29', 0)
EX4484 <- c('EX4484-52', 0)
both <- rbind(both, B1Sed)

no_IT_butCPA <- rbind(no_IT_butCPA, JACRDV01,Altiarchaeota, SpSt1190, Korarchaeota, Undinarchaeoata)
no_IT_butCPA$no_IT_butCPA <- as.numeric(no_IT_butCPA$no_IT_butCPA)

CPA_absent <- IT_Archaea[IT_Archaea$CPA_binary == 0,] 

no_CPA_but_IT <- aggregate(CPA_absent$IT_binary, by=list(Phyla=CPA_absent$Phyla), FUN=sum)
no_CPA_but_IT  <- no_CPA_but_IT  %>% rename(no_CPA_but_IT =x)


no_CPA_but_IT <- rbind(no_CPA_but_IT, JACRDV01, EX4484, Altiarchaeota, Nanohalarchaeota, Undinarchaeoata, B1Sed, Methanobacteriota_B, SpSt1190)


no_CPA_but_IT$no_CPA_but_IT <- as.numeric(no_CPA_but_IT$no_CPA_but_IT)
Phyla_A_no_AP_count <- aggregate(IT_Archaea$Antiporter_binary==0, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)

Phyla_A_nothing_count <- aggregate(IT_Archaea$Antiporter_binary==0, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
sum(Phyla_A_nothing_count$x)
Phyla_A_no_AP_count <- Phyla_A_no_AP_count %>% rename(total_noAP=x)


# counts of species with a specific antiproter per phyla
Phyla_A_AP_count <- aggregate(IT_Archaea$Antiporter_count, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_AP_count <- Phyla_A_AP_count %>% rename(total_AP=x)


Phyla_A_CPA_count <- aggregate(IT_Archaea$CPA_count, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_CPA_count <- Phyla_A_CPA_count %>% rename(total_CPA=x)

Phyla_A_AP_species_count <- aggregate(IT_Archaea$Antiporter_binary, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_AP_species_count <- Phyla_A_AP_species_count %>% rename(species_with_AP=x)

Phyla_A_NhaB_species_count <- aggregate(IT_Archaea$NhaB_binary==1, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_NhaB_species_count <- Phyla_A_NhaB_species_count %>% rename(species_NhaB=x)

Phyla_A_NhaC_species_count <- aggregate(IT_Archaea$NhaC_binary==1, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_NhaC_species_count <- Phyla_A_NhaC_species_count %>% rename(species_NhaC=x)

Phyla_A_species_NhaD_count <- aggregate(IT_Archaea$NhaD_binary==1, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_species_NhaD_count <- Phyla_A_species_NhaD_count %>% rename(species_NhaD=x)


#total count of the CPA per phyla
Phyla_A_CPA_count <- aggregate(IT_Archaea$CPA_count, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_CPA_count <- Phyla_A_CPA_count %>% rename(total_CPA=x)

Phyla_A_NhaB_count <- aggregate(IT_Archaea$NhaB_count, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_NhaB_count <- Phyla_A_NhaB_count %>% rename(total_NhaB=x)

Phyla_A_NhaC_count <- aggregate(IT_Archaea$NhaC_count, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_NhaC_count <- Phyla_A_NhaC_count %>% rename(total_NhaC=x)


Phyla_A_NhaD_count <- aggregate(IT_Archaea$NhaD_count, by=list(Phyla=IT_Archaea$Phyla), FUN=sum)
Phyla_A_NhaD_count <- Phyla_A_NhaD_count %>% rename(total_NhaD=x)



```

```{r}
Phyla_A_observations <- IT_Archaea%>% count(Phyla)

#merge all previous dataframes into one
Phyla_A_accounts <- merge(Phyla_A_no_AP_count, Phyla_A_observations, by='Phyla')

Phyla_A_accounts <- merge(Phyla_A_accounts, no_IT_butCPA, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, both, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, no_CPA_but_IT, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, Phyla_A_AP_count, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, Phyla_A_AP_species_count, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, Phyla_A_CPA_count, by='Phyla')
Phyla_A_accounts$IT_count <- Phyla_A_accounts$total_AP - Phyla_A_accounts$total_CPA
Phyla_A_accounts <- merge(Phyla_A_accounts, Phyla_A_NhaB_species_count, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, Phyla_A_NhaC_species_count, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, Phyla_A_species_NhaD_count, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, Phyla_A_NhaB_count, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, Phyla_A_NhaC_count, by='Phyla')
Phyla_A_accounts <- merge(Phyla_A_accounts, Phyla_A_NhaD_count, by='Phyla')

#generate average number of genes per species with set gene
Phyla_A_accounts$CPA_per_species <- Phyla_A_accounts$total_CPA/Phyla_A_accounts$species_with_AP
Phyla_A_accounts$AP_per_species <- Phyla_A_accounts$total_AP/Phyla_A_accounts$species_with_AP

Phyla_A_accounts$onlyCPA <- 100*Phyla_A_accounts$no_IT_butCPA/Phyla_A_accounts$n
Phyla_A_accounts$bothCPA_and_IT <- 100*as.numeric(Phyla_A_accounts$both_CPA_IT)/Phyla_A_accounts$n
Phyla_A_accounts$onlyIT <- 100*Phyla_A_accounts$no_CPA_but_IT/Phyla_A_accounts$n

```


```{r,Archaea_plot}

Percentages_ArcPhyla <- Phyla_A_accounts[,c(1,19,20, 21)]
Percentages_ArcPhyla$Phyla <- as.factor(Percentages_ArcPhyla$Phyla)

pivot_percentages_MJTARC <- Percentages_ArcPhyla %>% pivot_longer(
  cols=-Phyla,
  names_to = 'status',
  values_to = 'percentage') 

#informed by GTDB226 phylogeny
Kingdoms_A <- c('Halobacteriota', 'Thermoplasmatota', 'Methanobacteriota', 'Hydrothermarchaeota', 'Methanobacteriota_B', 'JACRDV01', 'Hadarchaeota','Thermoproteota', 'Korarchaeota','Asgardarchaeota','Nanobdellota','Aenigmatarchaeota','Nanohalarchaeota','EX4484-52','SpSt-1190','Undinarchaeota', 'Micrarchaeota', 'Iainarchaeota', 'B1Sed10-29', 'Altiarchaeota')
pivot_percentages_MJTARC$status <- factor(pivot_percentages_MJTARC$status, levels=c( 'onlyIT', 'bothCPA_and_IT', 'onlyCPA' ))
pivot_percentages_MJTARC$Phyla <- factor(pivot_percentages_MJTARC$Phyla, levels=Kingdoms_A)

pivArc<-ggplot(pivot_percentages_MJTARC, aes(y = Phyla, x=percentage, fill=status, alpha=status))+ geom_bar(stat='identity') + scale_fill_manual(values = c('brown1', 'darkviolet', 'deepskyblue1')) + scale_alpha_manual(values = c(0.3, 0.9, 0.5)) + theme_minimal() + 
  theme(axis.text.x = element_text(color = "black", size = 25, angle = 0, hjust = .5, vjust = .5, face = "plain"),
        axis.text.y = element_text(color = "black", size = 40, angle = 0, hjust = 1, vjust = 0, face = "bold"),  
        axis.title.x = element_text(color = "black", size = 20, angle = 0, hjust = .5, vjust = 0, face = "plain"),
        axis.title.y = element_text(color = "black", size = 25, angle = 0, hjust = .5, vjust = .5, face = "bold")) +theme(axis.text=element_text(size=30), legend.text = element_text(size=25)) + ggtitle('Archaea') +
  theme(plot.title = element_text(size = 50, face='bold' ),
        plot.subtitle = element_text(size = 20)) + theme(legend.text = element_text(size=40), legend.title = element_text(size=50, face='bold'))
pivArc
```

```{r}
OrderA_Phyla <-  mergedA_all %>% arrange(desc(average_Both))
List_PhylaA <- OrderA_Phyla$Phyla
mergedA_all <- mergedA_all[c('Phyla', 'average_CPA', 'average_Both', 'average_IT')]
pivot_Arc<- mergedA_all %>% pivot_longer(cols=-Phyla, names_to = 'status',values_to = 'Count') 

pivot_Arc$Phyla <- factor(pivot_Arc$Phyla, levels=c(List_PhylaA))
pivot_Arc$status <- factor(pivot_Arc$status, levels=c('average_CPA', 'average_Both', 'average_IT'))
PERA<-ggplot(pivot_Arc, aes(fill=status, y=Count, x=Phyla)) + geom_bar(position = 'dodge', stat='identity') + theme_minimal()  + scale_fill_manual(values = c( 'deepskyblue1', 'darkviolet','brown1')) + scale_alpha_manual(values = c(0.3, 0.9, 0.5)) +
  theme(axis.text.x = element_text(color = "black", size = 45, angle = 90, hjust = .5, vjust = .5, face = "bold"),
        axis.text.y = element_text(color = "black", size = 20, angle = 0, hjust = 1, vjust = 0.5, face = "bold"),  
        axis.title.x = element_text(color = "black", size = 20, angle = 0, hjust = .5, vjust = 0, face = "bold"),
        axis.title.y = element_text(color = "black", size = 25, angle = 0, hjust = .5, vjust = .5, face = "bold")) + geom_vline(xintercept = c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5,13.5,14.5,15.5,16.5,17.5,18.5, 19.5)) + geom_hline(yintercept = mean(IT_Archaea$Antiporter_count[IT_Archaea$Antiporter_binary==1]), lty='dashed') +
  scale_y_continuous(limits = c(0,8), expand = c(0, 0)) +
  labs(title = "Archaea",
       subtitle = "Average antiporter per species per phyla grouped by antiporter presence"
       )+
  theme(plot.title = element_text(size = 50, face='bold' ),
        plot.subtitle = element_text(size = 20)) +theme(legend.text = element_text(size=40), legend.title = element_text(size=50, face='bold'))

PERA
```



```{r load_Bacteria}
#these are the files containing all the antiporters, in our architecture it is called:
IT_Bacteria <- read.delim('/nesi/nobackup/uc04105/new_databases_May/GTDB_226/IT_Bacteria_7OKT.tsv', sep='\t', stringsAsFactors = TRUE)


IT_Bacteria %>% separate(GTDB_tax, c('Domain_Phyla', 'Lower_class'), ';c__') -> IT_Bacteria
IT_Bacteria %>% separate(Domain_Phyla, c('Domain', 'Phyla'), ';p__') ->IT_Bacteria
IT_Bacteria %>% separate(Lower_class, c('Class', 'Lower_order'), ';o__') -> IT_Bacteria
IT_Bacteria %>% separate(Lower_order, c('Order', 'Lower_family'), ';f__') -> IT_Bacteria
IT_Bacteria %>% separate(Lower_family, c('Family', 'Lower_genera'), ';g__') -> IT_Bacteria
IT_Bacteria %>% separate(Lower_genera, c('Genera', 'Species'), ';s__') -> IT_Bacteria



GTDB_Bac <- read.delim('/nesi/nobackup/uc04105/new_databases_May/GTDB_226/bac120_metadata.tsv', sep='\t', stringsAsFactors = TRUE, row.names = NULL)
GTDB_Bac$GTDB_id <- GTDB_Bac$accession

IT_Bacteria <- merge(IT_Bacteria, GTDB_Bac, by = 'GTDB_id')
#Antiporters_Bacteria$selected_cols_sum <- rowSums(Antiporters_Bacteria[, c("Kef", "CPA1", "CPA2","Uncharcterized", "NhaA")]) - Antiporters_Bacteria$CPA_count

IT_Bacteria <- IT_Bacteria[IT_Bacteria$Completeness> 90 & IT_Bacteria$Contamination < 5 & IT_Bacteria$trna_aa_count > 17,]

IT_Bacteria$Phyla[IT_Bacteria$Phyla == 'Bacteroidota_A'] <- 'Bacteroidota'
IT_Bacteria$Phyla[IT_Bacteria$Phyla == 'Desulfobacterota_B'] <- 'Desulfobacterota'
IT_Bacteria$Phyla[IT_Bacteria$Phyla == 'Desulfobacterota_I'] <- 'Desulfobacterota'
rm(GTDB_Bac)
```

```{r}
#the factorization of the binary CPA is donflip graphe to make it more intuitive for the figure

IT_Bacteria <- IT_Bacteria %>% rename(
    "Completeness(%)" = Completeness)
IT_Bacteria$CPA_presence[IT_Bacteria$CPA_binary ==0] <- 'Absent'
IT_Bacteria$CPA_presence[IT_Bacteria$CPA_binary ==1] <- 'Present'

IT_Bacteria$IT_count <- IT_Bacteria$NhaB_count + IT_Bacteria$NhaC_count + IT_Bacteria$NhaD_count
IT_Bacteria$Antiporter_count <- IT_Bacteria$CPA_count + IT_Bacteria$IT_count

IT_Bacteria$IT_binary[IT_Bacteria$IT_count == 0] <- 0
IT_Bacteria$IT_binary[IT_Bacteria$IT_count > 0] <- 1

IT_Bacteria$Antiporter_binary[IT_Bacteria$Antiporter_count == 0] <- 0
IT_Bacteria$Antiporter_binary[IT_Bacteria$Antiporter_count >0] <- 1



```





```{r Protein_data_wrangling_Bac}
IT_Bacteria$IT_count <- IT_Bacteria$NhaB_count + IT_Bacteria$NhaC_count + IT_Bacteria$NhaD_count
IT_Bacteria$Antiporter_count <- IT_Bacteria$CPA_count + IT_Bacteria$IT_count

IT_Bacteria$IT_binary[IT_Bacteria$IT_count == 0] <- 0
IT_Bacteria$IT_binary[IT_Bacteria$IT_count > 0] <- 1

IT_Bacteria$Antiporter_binary[IT_Bacteria$Antiporter_count == 0] <- 0
IT_Bacteria$Antiporter_binary[IT_Bacteria$Antiporter_count >0] <- 1


```




```{r, merge_phyla}
#to preserve original df prior to merging phyla


#based on RED values the phyla are subset into different versions of the same phyla, we overturn that in our analysis
IT_Bacteria$Phyla[IT_Bacteria$Phyla == 'Bacteroidota_A'] <- 'Bacteroidota'
IT_Bacteria$Phyla[IT_Bacteria$Phyla == 'Desulfobacterota_B'] <- 'Desulfobacterota'
IT_Bacteria$Phyla[IT_Bacteria$Phyla == 'Desulfobacterota_I'] <- 'Desulfobacterota'

```

```{r, subset a selection of phyla}
PhylaB_observations <- IT_Bacteria %>% count(Phyla)
top_Bac_phyla <- c('Acidobacteriota', 'Actinomycetota', 'Armatimonadota', 'Bacteroidota', 'Bacillota', 'Campylobacterota', 'Chloroflexota', 'Cyanobacteriota', 'Deinococcota', 'Desulfobacterota','Fusobacteriota', 'Gemmatimonadota' , 'Patescibacteriota', 'Planctomycetota', 'Pseudomonadota', 'Spirochaetota', 'Synergistota', 'Thermotogota','Verrucomicrobiota', 'Vulcanimicrobiota') 
PhylaB_observations <- IT_Bacteria %>% count(Phyla)
PhylaB_observations <- PhylaB_observations[PhylaB_observations$Phyla %in% top_Bac_phyla,]


PhylaB_CPAobservations <- IT_Bacteria[IT_Bacteria$CPA_binary == 1,] %>%count(Phyla)

PhylaB_observations <- IT_Bacteria %>%count(Phyla)

```



```{r, count number of species per phyla}
PhylaB_observations <- PhylaB_observations %>% rename('Species_total' = n)
#count NhaX/CPA genes per phyla
PhylaB_CPAobservations <- IT_Bacteria[IT_Bacteria$CPA_binary == 1,] %>%count(Phyla)
PhylaB_CPAobservations<- merge(PhylaB_observations, PhylaB_CPAobservations, phyla = 'Phyla')
PhylaB_CPAobservations$percentage <-  (PhylaB_CPAobservations$n/PhylaB_CPAobservations$Species_total)*100

#NhaB
PhylaB_NhaBobservations <- IT_Bacteria[IT_Bacteria$NhaB_binary == 1,] %>%count(Phyla)
PhylaB_NhaBobservations<- merge(PhylaB_observations, PhylaB_NhaBobservations, phyla = 'Phyla')
PhylaB_NhaBobservations$percentage <-  (PhylaB_NhaBobservations$n/PhylaB_NhaBobservations$Species_total)*100

#NhaC
PhylaB_NhaCobservations <- IT_Bacteria[IT_Bacteria$NhaC_binary == 1,] %>%count(Phyla)
PhylaB_NhaCobservations<- merge(PhylaB_observations, PhylaB_NhaCobservations, phyla = 'Phyla')
PhylaB_NhaCobservations$percentage <-  (PhylaB_NhaCobservations$n/PhylaB_NhaCobservations$Species_total)*100

#NhaD
PhylaB_NhaDobservations <- IT_Bacteria[IT_Bacteria$NhaD_binary == 1,] %>%count(Phyla)
PhylaB_NhaDobservations<- merge(PhylaB_observations, PhylaB_NhaDobservations, phyla = 'Phyla')
PhylaB_NhaDobservations$percentage <-  (PhylaB_NhaDobservations$n/PhylaB_NhaDobservations$Species_total)*100




```





```{r, wrangle_bac_data, no antiporters}


Phyla_B_no_AP_count <- aggregate(IT_Bacteria$Antiporter_binary==0, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)

sum(Phyla_B_no_AP_count$x)*100/nrow(IT_Bacteria) #13.17% of species miss an antiporter
Phyla_B_nothing_count <- aggregate(IT_Bacteria$Antiporter_binary==0, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)

Phyla_B_no_AP_count <- Phyla_B_no_AP_count[Phyla_B_no_AP_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_no_AP_count <- Phyla_B_no_AP_count %>% rename(
    Phy = Major_tax,
    total_noAP=x)

IT_present <- IT_Bacteria[IT_Bacteria$IT_binary == 1,] 

both <- aggregate(IT_present$CPA_binary, by=list(Major_tax=IT_present$Phyla), FUN=sum)
both <- both[both$Major_tax %in% top_Bac_phyla,]
both <- both %>% rename(
    Phyla = Major_tax,
    both_CPA_IT=x)


IT_absent <- IT_Bacteria[IT_Bacteria$IT_binary == 0,] 

no_IT_butCPA <- aggregate(IT_absent$CPA_binary, by=list(Major_tax=IT_absent$Phyla), FUN=sum)
no_IT_butCPA <- no_IT_butCPA[no_IT_butCPA$Major_tax %in% top_Bac_phyla,]
no_IT_butCPA <- no_IT_butCPA %>% rename(
    Phyla = Major_tax,
    no_IT_butCPA=x)

CPA_absent <- IT_Bacteria[IT_Bacteria$CPA_binary == 0,] 

no_CPA_but_IT <- aggregate(CPA_absent$IT_binary, by=list(Major_tax=CPA_absent$Phyla), FUN=sum)
no_CPA_but_IT <- no_CPA_but_IT [no_CPA_but_IT $Major_tax %in% top_Bac_phyla,]
no_CPA_but_IT  <- no_CPA_but_IT  %>% rename(
    Phyla = Major_tax,
    no_CPA_but_IT =x)
Deinococcota <- c('Deinococcota', '0')
no_CPA_but_IT <- rbind(no_CPA_but_IT, Deinococcota)

Phyla_B_no_AP_count <- aggregate(IT_Bacteria$Antiporter_binary==0, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)

Phyla_B_nothing_count <- aggregate(IT_Bacteria$Antiporter_binary==0, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
sum(Phyla_B_nothing_count$x)
Phyla_B_no_AP_count <- Phyla_B_no_AP_count[Phyla_B_no_AP_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_no_AP_count <- Phyla_B_no_AP_count %>% rename(Phyla = Major_tax,
    total_noAP=x)


#count species with antiporter
Phyla_B_AP_count <- aggregate(IT_Bacteria$Antiporter_count, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_AP_count <- Phyla_B_AP_count[Phyla_B_AP_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_AP_count <- Phyla_B_AP_count %>% rename(
    Phyla = Major_tax,
    total_AP=x)

#count species with NhaB
Phyla_B_NhaB_species_count <- aggregate(IT_Bacteria$NhaB_binary==1, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_NhaB_species_count <- Phyla_B_NhaB_species_count[Phyla_B_NhaB_species_count $Major_tax %in% top_Bac_phyla,]
Phyla_B_NhaB_species_count <- Phyla_B_NhaB_species_count %>% rename(
    Phyla = Major_tax,
    species_NhaB=x)

#count species with NhaC
Phyla_B_NhaC_species_count <- aggregate(IT_Bacteria$NhaC_binary==1, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_NhaC_species_count <- Phyla_B_NhaC_species_count[Phyla_B_NhaC_species_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_NhaC_species_count <- Phyla_B_NhaC_species_count %>% rename(
    Phyla = Major_tax,
    species_NhaC=x)

#count species with NhaD
Phyla_B_species_NhaD_count <- aggregate(IT_Bacteria$NhaD_binary==1, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_species_NhaD_count <- Phyla_B_species_NhaD_count[Phyla_B_species_NhaD_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_species_NhaD_count <- Phyla_B_species_NhaD_count %>% rename(
    Phyla = Major_tax,
    species_NhaD=x)

#count species with CPA
Phyla_B_CPA_count <- aggregate(IT_Bacteria$CPA_count, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_CPA_count <- Phyla_B_CPA_count[Phyla_B_CPA_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_CPA_count <- Phyla_B_CPA_count %>% rename(
    Phyla = Major_tax,
    total_CPA=x)


#count NhaB per phyla
Phyla_B_NhaB_count <- aggregate(IT_Bacteria$NhaB_count, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_NhaB_count <- Phyla_B_NhaB_count[Phyla_B_NhaB_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_NhaB_count <- Phyla_B_NhaB_count %>% rename(
    Phyla = Major_tax,
    total_NhaB=x)
#count NhaC per phyla
Phyla_B_NhaC_count <- aggregate(IT_Bacteria$NhaC_count, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_NhaC_count <- Phyla_B_NhaC_count[Phyla_B_NhaC_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_NhaC_count <- Phyla_B_NhaC_count %>% rename(
    Phyla = Major_tax,
    total_NhaC=x)
#count NhaD per phyla
Phyla_B_NhaD_count <- aggregate(IT_Bacteria$NhaD_count, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_NhaD_count <- Phyla_B_NhaD_count[Phyla_B_NhaD_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_NhaD_count <- Phyla_B_NhaD_count %>% rename(
    Phyla = Major_tax,
    total_NhaD=x)
#count CPA per phyla
Phyla_B_CPA_count <- aggregate(IT_Bacteria$CPA_count, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_CPA_count <- Phyla_B_CPA_count[Phyla_B_CPA_count$Major_tax %in% top_Bac_phyla,]
Phyla_B_CPA_count <- Phyla_B_CPA_count %>% rename(
    Phyla = Major_tax,
    total_CPA=x)
#count Anitporters per phyla
Phyla_B_AP_species_count <- aggregate(IT_Bacteria$Antiporter_binary, by=list(Major_tax=IT_Bacteria$Phyla), FUN=sum)
Phyla_B_AP_species_count <- Phyla_B_AP_species_count[Phyla_B_AP_species_count $Major_tax %in% top_Bac_phyla,]
Phyla_B_AP_species_count <- Phyla_B_AP_species_count %>% rename(
    Phyla = Major_tax,
    species_with_AP=x)

#merge data
Phyla_B_observations <- IT_Bacteria[IT_Bacteria$Phyla %in% top_Bac_phyla,] %>% count(Phyla)
Phyla_B_accounts <- merge(Phyla_B_no_AP_count, Phyla_B_observations, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, no_IT_butCPA, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, both, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, no_CPA_but_IT, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, Phyla_B_AP_count, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, Phyla_B_AP_species_count, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, Phyla_B_CPA_count, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, Phyla_B_NhaB_species_count, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, Phyla_B_NhaC_species_count, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, Phyla_B_species_NhaD_count, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, Phyla_B_NhaB_count, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, Phyla_B_NhaC_count, by='Phyla')
Phyla_B_accounts <- merge(Phyla_B_accounts, Phyla_B_NhaD_count, by='Phyla')

Phyla_B_accounts$IT_count <- Phyla_B_accounts$total_AP - Phyla_B_accounts$total_CPA

#generate average number of genes per species who got me
Phyla_B_accounts$CPA_per_species <- Phyla_B_accounts$total_CPA/Phyla_B_accounts$species_with_AP
Phyla_B_accounts$NhaB_per_species <- Phyla_B_accounts$total_NhaB/Phyla_B_accounts$species_NhaB
Phyla_B_accounts$NhaC_per_species <- Phyla_B_accounts$total_NhaC/Phyla_B_accounts$species_NhaC
Phyla_B_accounts$NhaD_per_species <- Phyla_B_accounts$total_NhaD/Phyla_B_accounts$species_NhaD


Phyla_B_accounts$AP_per_species <- Phyla_B_accounts$total_AP/Phyla_B_accounts$species_with_AP

Phyla_B_accounts$onlyCPA <- 100*Phyla_B_accounts$no_IT_butCPA/Phyla_B_accounts$n
Phyla_B_accounts$bothCPA_and_IT <- 100*Phyla_B_accounts$both_CPA_IT/Phyla_B_accounts$n
Phyla_B_accounts$onlyIT <- 100*as.numeric(Phyla_B_accounts$no_CPA_but_IT)/Phyla_B_accounts$n

Phyla_B_accounts$CPA_presence <- (Phyla_B_accounts$both_CPA_IT + Phyla_B_accounts$no_IT_butCPA)/Phyla_B_accounts$n



```

```{r percentages-bac}

Percentages_BacPhyla <- Phyla_B_accounts[,c(1,22,23,24)]
Percentages_BacPhyla$Phyla <- as.factor(Percentages_BacPhyla$Phyla)

pivot_percentages_MJTBAC <- Percentages_BacPhyla %>% pivot_longer(
  cols=-Phyla,
  names_to = 'status',
  values_to = 'percentage') 

#kingdoms are those divined by Goker and oren et al.2024 (see literature reference Oren)
Kingdoms_B <- c('Acidobacteriota', 'Bacteroidota','Campylobacterota' ,'Desulfobacterota','Gemmatimonadota', 'Planctomycetota', 'Pseudomonadota','Spirochaetota','Verrucomicrobiota','Actinomycetota','Armatimonadota','Bacillota','Chloroflexota','Cyanobacteriota','Deinococcota','Synergistota','Thermotogota','Fusobacteriota','Patescibacteriota','Vulcanimicrobiota')


pivot_percentages_MJTBAC$status <- factor(pivot_percentages_MJTBAC$status, levels=c( 'onlyIT', 'bothCPA_and_IT', 'onlyCPA' ))
pivot_percentages_MJTBAC$Phyla <- factor(pivot_percentages_MJTBAC$Phyla, levels=Kingdoms_B) 
pivB<-ggplot(pivot_percentages_MJTBAC, aes(y = Phyla, x=percentage, fill=status, alpha=status))+ geom_bar(stat='identity') + scale_fill_manual(values = c('brown1', 'darkviolet', 'deepskyblue1')) + scale_alpha_manual(values = c(0.3, 0.9, 0.5)) + theme_minimal() + 
  theme(axis.text.x = element_text(color = "black", size = 25, angle = 0, hjust = .5, vjust = .5, face = "plain"),
        axis.text.y = element_text(color = "black", size = 40, angle = 0, hjust = 1, vjust = 0, face = "bold"),  
        axis.title.x = element_text(color = "black", size = 20, angle = 0, hjust = .5, vjust = 0, face = "plain"),
        axis.title.y = element_text(color = "black", size = 25, angle = 0, hjust = .5, vjust = .5, face = "bold")) +theme(axis.text=element_text(size=30), legend.text = element_text(size=25))  + ggtitle('Bacteria') +
  theme(plot.title = element_text(size = 50, face='bold' ),
        plot.subtitle = element_text(size = 20)) +
  theme(plot.title = element_text(size = 50, face='bold' ),
        plot.subtitle = element_text(size = 20)) + theme(legend.text = element_text(size=40), legend.title = element_text(size=50, face='bold'))
pivB
```







```{r, Bacteria with both CPA and IT}
Both_BONLY_SPECIES <- IT_Bacteria[IT_Bacteria$IT_count >0 & IT_Bacteria$CPA_count > 0 & IT_Bacteria$Phyla %in% top_Bac_phyla,]


Observations_BBoth_only<- aggregate(as.numeric(Both_BONLY_SPECIES$Antiporter_count), by=list(Major_tax=Both_BONLY_SPECIES$Phyla), FUN=sum)
Observations_BBoth_only <- Observations_BBoth_only %>% rename(
    Phyla = Major_tax,
    total_Both=x)



Both_BONLY_Species_count <- Both_BONLY_SPECIES %>% count(Phyla)
Both_BONLY_Species_count_per_phyla <- Both_BONLY_SPECIES %>% count(Phyla)

rm(Both_BONLY_SPECIES)
rm(Both_BONLY_Species_count)

Both_BONLY_Species_count_per_phyla <- merge(Both_BONLY_Species_count_per_phyla, Observations_BBoth_only, phyla = 'Phyla')
```

```{r, complete_cpa_only}
CPA_BONLY_SPECIES <- IT_Bacteria[IT_Bacteria$IT_count == '0' & IT_Bacteria$CPA_count > 0 & IT_Bacteria$Phyla %in% top_Bac_phyla,]


Observations_cpa_Bonly<- aggregate(as.numeric(CPA_BONLY_SPECIES$CPA_count), by=list(Major_tax=CPA_BONLY_SPECIES$Phyla), FUN=sum)
Observations_cpa_Bonly <- Observations_cpa_Bonly %>% rename(
    Phyla = Major_tax,
    total_CPA=x)

CPA_BONLY_Species_count <- CPA_BONLY_SPECIES %>% count(Phyla)
CPA_BONLY_Species_count_per_phyla <- CPA_BONLY_SPECIES %>% count(Phyla)

rm(CPA_BONLY_SPECIES)
rm(CPA_BONLY_Species_count)

CPA_BONLY_Species_count_per_phyla <- merge(CPA_BONLY_Species_count_per_phyla, Observations_cpa_Bonly, phyla = 'Phyla')

Missing_phyla_list = list(c())
Missing_N_list = list(c())
Missing_gene_count = list(c())
for (taxa in PhylaB_observations$Phyla) {

  if (taxa %!in% CPA_BONLY_Species_count_per_phyla$Phyla) {
    print(taxa)
    Missing_phyla_list<-append(Missing_phyla_list, taxa)
    Missing_N_list<-append(Missing_N_list, 1000)
    Missing_gene_count<-append(Missing_gene_count, 0.001)
  }
}


Phyla<- unlist(Missing_phyla_list, use.names=FALSE)
n<-unlist(Missing_N_list, use.names = FALSE)
total_CPA <-unlist(Missing_gene_count, use.names =FALSE)
Missingdata <- data.frame(Phyla, n, total_CPA)

CPA_BONLY_Species_count_per_phyla <- rbind(CPA_BONLY_Species_count_per_phyla, Missingdata)
```


```{r, Species with IT only Bacteria}
IT_BONLY_SPECIES <- IT_Bacteria[IT_Bacteria$IT_count >0 & IT_Bacteria$CPA_count=='0' & IT_Bacteria$Phyla %in% top_Bac_phyla,]




Observations_BIT_only<- aggregate(as.numeric(IT_BONLY_SPECIES$IT_count), by=list(Major_tax=IT_BONLY_SPECIES$Phyla), FUN=sum)
Observations_BIT_only <- Observations_BIT_only %>% rename(
    Phyla = Major_tax,
    total_IT=x)

IT_BONLY_Species_count <- IT_BONLY_SPECIES %>% count(Phyla)
IT_BONLY_Species_count_per_phyla <- IT_BONLY_SPECIES %>% count(Phyla)

rm(IT_BONLY_SPECIES)
rm(IT_BONLY_Species_count)

IT_BONLY_Species_count_per_phyla <- merge(IT_BONLY_Species_count_per_phyla, Observations_BIT_only, phyla = 'Phyla')

Missing_phyla_list = list(c())
Missing_N_list = list(c())
Missing_gene_count = list(c())
for (taxa in PhylaB_observations$Phyla) {

  if (taxa %!in% IT_BONLY_Species_count_per_phyla$Phyla) {
    print(taxa)
    Missing_phyla_list<-append(Missing_phyla_list, taxa)
    Missing_N_list<-append(Missing_N_list, 1000)
    Missing_gene_count<-append(Missing_gene_count, 0.001)
  }
}


Phyla<- unlist(Missing_phyla_list, use.names=FALSE)
n<-unlist(Missing_N_list, use.names = FALSE)
total_IT <-unlist(Missing_gene_count, use.names =FALSE)
Missingdata <- data.frame(Phyla, n, total_IT)

IT_BONLY_Species_count_per_phyla <- rbind(IT_BONLY_Species_count_per_phyla, Missingdata)
```

```{r, merge all the Bacteria data}
mergedB_all <-merge(Both_BONLY_Species_count_per_phyla,CPA_BONLY_Species_count_per_phyla, by='Phyla')
mergedB_all <- merge(mergedB_all,IT_BONLY_Species_count_per_phyla, by='Phyla')

mergedB_all <- mergedB_all %>% rename(
    n_species_onlyIT = n,
    n_species_both = n.x,
    n_species_onlyCPA = n.y)
mergedB_all$average_CPA <- mergedB_all$total_CPA/mergedB_all$n_species_onlyCPA
mergedB_all$average_Both <- mergedB_all$total_Both/mergedB_all$n_species_both
mergedB_all$average_IT <- mergedB_all$total_IT/mergedB_all$n_species_onlyIT
```

```{r, reorder Bacteria and generate figure}
order_byBPhyla <-  mergedB_all %>% arrange(desc(average_Both))
List_PhylaB <- order_byBPhyla$Phyla
mergedB_all <- mergedB_all[c('Phyla', 'average_CPA', 'average_Both', 'average_IT')]

pivot_Bac<- mergedB_all %>% pivot_longer(cols=-Phyla, names_to = 'status',values_to = 'Count') 

pivot_Bac$Phyla <- factor(pivot_Bac$Phyla, levels=c(List_PhylaB))

pivot_Bac$status <- factor(pivot_Bac$status, levels=c('average_CPA', 'average_Both', 'average_IT'))
PERB<- ggplot(pivot_Bac, aes(fill=status, y=Count, x=Phyla)) + geom_bar(position = 'dodge', stat='identity') + theme_minimal()  + scale_fill_manual(values = c( 'deepskyblue1', 'darkviolet','brown1')) + scale_alpha_manual(values = c(0.3, 0.9, 0.5)) +
  theme(axis.text.x = element_text(color = "black", size = 45, angle = 90, hjust = .5, vjust = .5, face = "bold"),
        axis.text.y = element_text(color = "black", size = 20, angle = 0, hjust = 1, vjust = 0.5, face = "bold"),  
        axis.title.x = element_text(color = "black", size = 20, angle = 0, hjust = .5, vjust = 0, face = "plain"),
        axis.title.y = element_text(color = "black", size = 25, angle = 0, hjust = .5, vjust = .5, face = "bold")) + geom_vline(xintercept = c(1.5,2.5,3.5,4.5,5.5,6.5,7.5,8.5,9.5,10.5,11.5,12.5,13.5,14.5,15.5,16.5,17.5,18.5, 19.5)) + geom_hline(yintercept = mean(IT_Bacteria$Antiporter_count[IT_Bacteria$Antiporter_binary==1]), lty='dashed') +
  scale_y_continuous(limits = c(0,9), expand = c(0, 0)) +
  labs(title = "Bacteria",
       subtitle = "Average antiporter per species grouped by antiporter presence"
       )+
  theme(plot.title = element_text(size = 50, face='bold' ),
        plot.subtitle = element_text(size = 20)) + theme(legend.text = element_text(size=40), legend.title = element_text(size=50, face='bold')) 
PERB
```





```{r, arrange_archaea_plot}

library(gridExtra)

pdf(file = "Archaea.pdf",   # The directory you want to save the file in
    width = 60, # The width of the plot in inches
    height = 30) 

pivArc

dev.off()
```

```{r, arrange_archaea_plot}

library(gridExtra)

png(file = "Archaea.png",   # The directory you want to save the file in
    width = 8000, # The width of the plot in inches
    height = 4000) 

pivArc

dev.off()
```

```{r, arrange_bacteria_plot}

library(gridExtra)

pdf(file = "Bacteria.pdf",   # The directory you want to save the file in
    width = 60, # The width of the plot in inches
    height = 30) 

pivB

dev.off()
```

```{r, arrange_archaea_plot}

library(gridExtra)

png(file = "Bacteria.png",   # The directory you want to save the file in
    width = 8000, # The width of the plot in inches
    height = 4000) 

pivB
dev.off()
```


```{r, arrange_euk_plot}

library(gridExtra)

pdf(file = "Eukarya.pdf",   # The directory you want to save the file in
    width = 30, # The width of the plot in inches
    height = 30) 

pivE

dev.off()
```


```{r, arrange_archaea_plot}

library(gridExtra)

png(file = "Eukarya.png",   # The directory you want to save the file in
    width = 8000, # The width of the plot in inches
    height = 4000) 

pivE

dev.off()
```


```{r, arrange_archaea_plot}

library(gridExtra)

png(file = "Eukaryacount.png",   # The directory you want to save the file in
    width = 8000, # The width of the plot in inches
    height = 4000) 

PERE
dev.off()
```

```{r, arrange_archaea_plot}

library(gridExtra)

png(file = "Aerchaeacount.png",   # The directory you want to save the file in
    width = 8000, # The width of the plot in inches
    height = 4000) 

PERA
dev.off()
```

```{r, arrange_archaea_plot}

library(gridExtra)

png(file = "Bacteriacount.png",   # The directory you want to save the file in
    width = 8000, # The width of the plot in inches
    height = 4000) 

PERB
dev.off()
```

```{r, arrange_figure}

library(gridExtra)

pdf(file = "ALL.pdf",   # The directory you want to save the file in
    width = 90, # The width of the plot in inches
    height = 40) 

MF<-grid.arrange(PERB, PERA, PERE, nrow = 1)
MF

dev.off()
```

```{r, arrange_figure}

library(gridExtra)

png(file = "All.png",   # The directory you want to save the file in
    width = 12000, # The width of the plot in inches
    height = 6000) 

MF<-grid.arrange(PERB, PERA, PERE, nrow = 1)
MF

dev.off()
```

```{r, supplementary figure of distribution of number of species}
dev.off()
par(mfrow=c(7,2))
#a bit differently generated but this allows me to compare the distirbution of gene count per species.
#i.e how many species have 1 CPA, vs how many species have 2 CPA etc etc etc
Psuedomona_edit <- IT_Bacteria[IT_Bacteria$Phyla %in% Pseudomonadati,]
Thermotogota_edit <- IT_Bacteria[IT_Bacteria$Phyla %in% Thermotogati,]
Bacilati_edit <- IT_Bacteria[IT_Bacteria$Phyla %in% Bacilati,]
Patesci_edit <- IT_Bacteria[IT_Bacteria$Phyla == 'Patescibacteriota',]

hist(Psuedomona_edit$Antiporter_count[Psuedomona_edit$Antiporter_count > 0], breaks='Scott', col ='pink', main = 'Pseudo distribution antiporter genes')
hist(Psuedomona_edit$CPA_count[Psuedomona_edit$CPA_count > 0], breaks='Scott', col='deepskyblue', main = 'Pseudo distribution CPA genes')

hist(Thermotogota_edit$Antiporter_count[Thermotogota_edit$Antiporter_count > 0], breaks='Scott', col ='coral1', main = 'Thermotogota distribution antiporter genes')
hist(Thermotogota_edit$CPA_count[Thermotogota_edit$CPA_count > 0], breaks='Scott', col='aquamarine', main = 'Thermotogota distribution CPA genes')

hist(Bacilati_edit$Antiporter_count[Bacilati_edit$Antiporter_count > 0], breaks='Scott', col ='brown', main = 'Bacilati distribution antiporter genes')
hist(Bacilati_edit$CPA_count[Bacilati_edit$CPA_count > 0], breaks='Scott', col='cadetblue', main = 'Bacilati distribution CPA genes')

hist(Patesci_edit$Antiporter_count[Patesci_edit$Antiporter_count > 0], breaks='Scott', col ='chocolate', main = 'Patesci distribution antiporter genes')
hist(Patesci_edit$CPA_count[Patesci_edit$CPA_count > 0], breaks='Scott', col='darkslategray', main = 'Patesci distribution CPA genes')

#Archaea
hist(Halo$Antiporter_count[Halo$Antiporter_count > 0], breaks='Scott', col ='orange', main = 'Halo distribution antiporter genes')
hist(Halo$CPA_count[Halo$CPA_count > 0], breaks='Scott', col='blue', main = 'Halo distribution CPA genes')

hist(TACK$Antiporter_count[TACK$Antiporter_count > 0], breaks='Scott', col='red', main = 'TACK distribution antiporter genes')
hist(TACK$CPA_count[TACK$CPA_count > 0], breaks='Scott', col='cyan', main = 'TACK distribution CPA genes')

hist(Nano$Antiporter_count[Nano$Antiporter_count > 0], breaks='Scott', col='brown1', main = 'Nano distribution antiporter genes')
hist(Nano$CPA_count[Nano$CPA_count > 0], breaks='Scott', col ='azure', main = 'Nano distribution CPA genes')
```
